### PR TITLE
[expo-localization] Remove deprecated methods from mocks

### PR DESCRIPTION
**Pull request by aleqsio**
# Why

This PR just removes some methods from docs that we deprecated a few months ago. It makes the docs clearer and less cluttered.

# How

# Test Plan

# Checklist

- [ ] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [ ] Conforms with the [Documentation Writing Style Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)
- [ ] This diff will work correctly for `npx expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

File: packages/expo-localization/src/Localization.ts [hunk 1]
```diff
@@ -62,7 +68,7 @@ export const isRTL = ExpoLocalization.isRTL;
 
 // @needsAudit
 /**
- * Consider using Localization.getLocales() for a list of user preferred locales instead.
+ * @deprecated Use Localization.getLocales() instead.
```
Comment (amandeepmittal) [inline packages/expo-localization/src/Localization.ts:71]:
```suggestion
 * @deprecated Use `Localization.getLocales()` instead.
```

**Review by amandeepmittal — APPROVED** [review]
One small suggestion to add inline syntax highlight but rest lgtm! Thank you for cleaning this!

File: packages/expo-localization/src/Localization.ts (hunk 1 above)
Comment (Simek) [reply packages/expo-localization/src/Localization.ts:71]: We should also link the recommended replacement in the notice, if possible.

Comment (amandeepmittal) [reply packages/expo-localization/src/Localization.ts:71]: Yes, great point. I just updated the suggestion.

**Pull request merged**